src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/AbstractNameCheck.java

```diff
@@ -66,6 +66,7 @@ protected AbstractNameCheck(String format) {
      * Set the format for the specified regular expression.
      *
      * @param pattern the new pattern
+     * @since 3.0
      */
     public final void setFormat(Pattern pattern) {
```

Comment:
Class existed in 3.0 @ https://github.com/checkstyle/checkstyle/blob/release3_0/src/checkstyle/com/puppycrawl/tools/checkstyle/checks/AbstractNameCheck.java

But I don't see the format setter.

Reply:
If the list of tags is to be believed correct, the previous version is 2.4 . There was no named checks in 2.4.
https://github.com/checkstyle/checkstyle/tree/release2_4/src/checkstyle/com/puppycrawl/tools/checkstyle

But the method was not in the specific abstract at the version defined. Since it wasn't in 3, it was moved there after 3. The question is which version should be documented.

Reply:
from Java perspective setFormat appeared in class at 6.14, so if we remoe xdoc from consideration for a second, and look at this problem from pure java+javadoc position, version must be 6.14. 
https://github.com/checkstyle/checkstyle/issues/2451
https://github.com/checkstyle/checkstyle/commit/541d6e2581aaa22c98059b9e82a167ee4dced4a7#diff-0b9b4ab5f7aaa48aa97454c7428978d66b4889bc3a3f12eaa8d10db9c54eaf95R60

for Module/Check it appeared in dark time of 2.4-3.0, I am not sure we cat trust tags of that time, it might be damaged during migration from from CVS to SVN to GIT.
Back in time of xdoc rules introduction, we did simple rule, if version is going before 3.0 and become hard to track, put 3.0.

I recommend to not be very accurate here, as nobody care, and we do not support such old versions.

Reply:
oldest web doc we have is https://checkstyle.sourceforge.io/version/4.4/ , no html for 5.0
what we have:
```
htdocs-4.4
htdocs-5.5
htdocs-5.6
htdocs-5.7
htdocs-5.8
htdocs-5.9
htdocs-6.0
....
```

Reply:
We can't really rely on web documentation as it was all manually updated before I came alone, and I had to make many fixes.

> from Java perspective setFormat appeared in class at 6.14, so if we remoe xdoc from consideration for a second, and look at this problem from pure java+javadoc position, version must be 6.14.
https://github.com/checkstyle/checkstyle/commit/541d6e2581aaa22c98059b9e82a167ee4dced4a7#diff-0b9b4ab5f7aaa48aa97454c7428978d66b4889bc3a3f12eaa8d10db9c54eaf95R60

After talking with @nrmancuso on this, we believe the documentation should say 6.14, when the method became part of this abstract class.

Reply:
In this case, it will affect xdoc, and it will be wrong in xdoc, affected Checks exited far before 6.14.
It is better to hardcode all such check+ property in our hack map.
In separate issue we can rename all `@since` to `@xdocsince`. And use it for xdoc purposes and do not mess with pure java/javadoc since

Reply:
You will need to use `SINCE_VERSION_FOR_INHERITED_PROPERTY` to resolve discrepancies between abstracts and xdocs. I mentioned before the calculation being used in XDoc macro for version number isn't right. It does not compare value between abstract method and check's since version. Either the code needs to be fixed or `SINCE_VERSION_FOR_INHERITED_PROPERTY` needs to be used.